compiler/src/dotty/tools/backend/jvm/BCodeHelpers.scala

```diff
@@ -926,7 +926,7 @@ trait BCodeHelpers extends BCodeIdiomatic with BytecodeWriters {
     // (one that doesn't erase to the actual signature). See run/t3452b for a test case.
 
     val memberTpe = atPhase(erasurePhase) { moduleClass.denot.thisType.memberInfo(sym) }
-    val erasedMemberType = TypeErasure.fullErasure(memberTpe)
+    val erasedMemberType = TypeErasure.transformInfo(sym, memberTpe)
```

Comment:
To deal with value classes we also need to strip out `ErasedValueType` anywhere in the type, ElimErasedValueType#elimEVT does that, it's an instance method but it could be moved to the companion.

Reply:
Thanks, added this change and a test for it.